xds/src/test/java/io/grpc/xds/XdsCommsTest.java

```diff
@@ -472,25 +480,49 @@ public void adsRpcRetry() {
 
     // The 6th RPC fails with response observer onError() without receiving initial response
     responseWriter.onError(new Exception("fake error"));
-    inOrder.verify(adsStreamCallback).onError();
+    inOrder.verify(endpointWatcher).onError(any(Status.class));
 
     // Retry is scheduled
     assertEquals(1, fakeClock.numPendingTasks(LB_RPC_RETRY_TASK_FILTER));
 
     // Shutdown cancels retry
-    xdsComms.shutdownLbRpc();
+    xdsComms.shutdown();
     assertEquals(0, fakeClock.numPendingTasks(LB_RPC_RETRY_TASK_FILTER));
   }
 
   @Test
   public void refreshAdsStreamCancelsExistingRetry() {
     responseWriter.onError(new Exception("fake error"));
-    verify(adsStreamCallback).onError();
+    verify(endpointWatcher).onError(any(Status.class));
     assertEquals(1, fakeClock.numPendingTasks(LB_RPC_RETRY_TASK_FILTER));
 
     xdsComms.refreshAdsStream();
     assertEquals(0, fakeClock.numPendingTasks(LB_RPC_RETRY_TASK_FILTER));
 
-    xdsComms.shutdownLbRpc();
+    xdsComms.shutdown();
+  }
+
+  private static XdsClient.EndpointUpdate endpointUpdatefromClusterAssignment(
```

Comment:
As mentioned above, can you create a new helper class for calculating LocalityLbEndpointsMap which stays thru your whole refactoring effort and once the dust settles you will eliminate that by refactoring?

Reply:
nit: name method with verb+noun.

Reply:
Done.